src/utils/curriculum/filtering.ts

```diff
@@ -153,18 +151,22 @@ export function useFilters(
     }
   }, [searchParams, defaultFilter]);
 
-  const setFilters = (newFilters: CurriculumFilters) => {
-    if (isCurricRoutingEnabled()) {
-      const url =
-        location.pathname +
-        "?" +
-        new URLSearchParams(
-          Object.entries(filtersToQuery(newFilters, defaultFilter)),
-        ).toString();
-      router.replace(url, undefined, { shallow: true });
-    }
-    setLocalFilters(newFilters);
-  };
+  const setFilters = useCallback(
+    (newFilters: CurriculumFilters) => {
+      if (isCurricRoutingEnabled()) {
+        const url =
+          location.pathname +
+          "?" +
+          new URLSearchParams(
+            Object.entries(filtersToQuery(newFilters, defaultFilter)),
+          ).toString();
+        // Use replaceState directly to avoid triggering router events
+        window.history.replaceState({}, "", url);
```

Comment:
Does `{shallow: true}` not achieve this? I wonder if `useCallback(...)` alone fixes the issue